### ISSUE TITLE
[enc2] Download versions extremely slow

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#15970]
### Steps
1. Enable encryption + default encryption module
2. Log out, log int
3. Create a file "versions.txt"
4. Edit and save it several times with the text editor to create versions
5. Try downloading versions

### Expected result
Quick download

### Actual result
Very slow with Chrome
(I haven't tried curl)

Possibly related to the file size issue ?

@DeepDiver1975 @schiesbn @th3fallen

**Comment by PVince81** [comment]
forgot to mention you too @jknockaert :smile: 

I suspect it could be similar to the bug we had with download of regular files.
I'll try and provide more info later (curl calls with timings, etc)

**Comment by PVince81** [comment]
When downloading a specific version, the size doesn't match:
```
% curl -D - 'http://root:admin@localhost/owncloud/index.php/apps/files_versions/download.php?file=%2Fversions.txt&revision=1430408165'
HTTP/1.1 200 OK
Date: Thu, 30 Apr 2015 15:36:51 GMT
Server: Apache/2.4.12 (Linux/SUSE)
X-Powered-By: PHP/5.6.7
Set-Cookie: ocsyk7ry70q6=5ihmpblr2t1vovuhd2k71hndqssk8enea6t1r4th1fesso9l8uu0; path=/owncloud; HttpOnly
Expires: 0
Cache-Control: must-revalidate, post-check=0, pre-check=0
Pragma: public
Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *; connect-src *
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: Sameorigin
X-Robots-Tag: none
Set-Cookie: ocsyk7ry70q6=pmerbp422t1i3p1h7c5pd1ic4cei4psu3qab0elp85dcc2g8s5j0; path=/owncloud; HttpOnly
Content-Disposition: attachment; filename*=UTF-8''versions.txt; filename="versions.txt"
Content-Length: 8244
Content-Type: text/plain;charset=UTF-8

curl: (18) transfer closed with 8225 bytes remaining to read
bbbbbbbbbbbbbbbbbbb
```

The "bbbb" is the actual content of the file, which is only a few bytes big.

8k looks like it's the encrypted size (with the first 8k header block)

**Comment by PVince81** [comment]
I debugged a bit and saw that `$this->filesize()` expects the field "encrypted" to be true in the database, which isn't the case for version files:
```
MariaDB [owncloud]> select fileid,path,encrypted,size from oc_filecache where path like 'files_versions%' order by path;
+--------+-----------------------------------------+-----------+-------+
| fileid | path                                    | encrypted | size  |
+--------+-----------------------------------------+-----------+-------+
|     17 | files/versions.txt                      |         1 |    10 |
|     24 | files_versions                          |         0 | 24692 |
|     28 | files_versions/versions.txt.v1430409043 |         0 |  8220 |
|     31 | files_versions/versions.txt.v1430409045 |         0 |  8232 |
|     34 | files_versions/versions.txt.v1430409048 |         0 |  8240 |
+--------+-----------------------------------------+-----------+-------+
5 rows in set (0.00 sec)
```

Also, I remember @SergioBertolinSG asking whether versions that existed before enabling encryption would still be unencrypted on disk. This would be achieved by having the old versions have the "encryption" flag set to 0, but the new ones set to 1.

**Comment by MorrisJobke** [comment]
Tested this too and it is fixed with #16070
